### ISSUE TITLE
Performance slow

### ISSUE THREAD

**Issue by ccharlton** [nex3/firesass#18]
When I enable FireSass I notice my browser really starts to crawl. When I disable FireSass then the browser responds as expected.

Great plugin! Just wanted to point out there's some performance issues, especially on a lower Dual-Core i5 CPU & <4GB RAM machine.

**Comment by aghouseh** [comment]
Yeah, I've got a new MacBook Pro Retina, and switching Firebug tabs with FireSass enabled takes 2-3 seconds; something seems amiss.

**Comment by ccharlton** [comment]
Ditto. MacBook Pro Retina 16GB RAM, 2.3GHz i7 CPU; Latest version of Firefox/Firebug & OS X.

**Comment by Timoti** [comment]
Having the same issue on OS 10.6, FF 14-18, Firebug 1.11.0, Firesass 0.012
And I have reset Firefox.

This combination worked before but doesnt seem to know

This tool is utterly essential for me and I just feel crippled. There are many like me who would be happy to help support this I'm sure - this wonderful extension really needs a little love.

In the meantime it would helpful to have some notes on what combo of FF, FB versions will work with FS

Right now I don't care it the set up is old, i just need it to work!

Please?

**Comment by cfg** [comment]
Same problem on Win7 x64. Downgrading to [Firefox 17.0.1](https://ftp.mozilla.org/pub/mozilla.org/firefox/releases/17.0.1/) and [Firebug 1.10.5](https://getfirebug.com/releases/firebug/1.10/) has made this mostly tolerable. Any later release of Firefox including 17.0.2 ESR seems to have the performance issues.

**Comment by Timoti** [comment]
hmm - I'm back still trying to get a handle on this  - have just tried cfg's recipe (though on OS X) and still ahve a 3 second lag for every click in firebug  - with firesass. It's fine without firesass running, and I recently started  with a fresh FF profile and am only running firebug and firesass.

I became utterly dependent on this tool but now coding has become an incredibly frustrating, hand clenching experience.

There's gotta be a way to fix this!

**Comment by ccharlton** [comment]
I wonder if the performance issue comes from a large amount of nodes being discovered & parsed. I haven't dug into the source myself yet, but I'm a little surprised more people haven't reported the same issue.

**Comment by lolmaus** [comment]
For me, FireSass makes Firebug's "Computed" tab freeze the browser for 5—10 seconds. Whenever i open the "Computed" tab, it freezes for a while. Surfing with this tab open makes Firefox constantly freeze.

**Comment by spoutnikz** [comment]
I think slow perfs on the "Computed" tab is directly linked to the stylesheet size. We have a css merger who aggregates all css in one and it has difficult when it is in this mode but no problem if stylesheets are short (even if there are many)

**Comment by lolmaus** [comment]
The freeze time of the Computed tab is indeed proportional to the size of the CSS file.

But the main reason of the freeze is the `:debug_info` switch of SASS. With debug info disabled, opening the Computed tab takes only 2 or 3 seconds. When i recompile the CSS with debug info enabled, opening the Computed tab of the same page takes 15 seconds to unfreeze.

**Comment by ccharlton** [comment]
No thoughts on what may help this? LibSass has helped compile times, but parsing is another rabbit hole.
